### PR TITLE
Release 12.1.0

### PR DESCRIPTION
**Pull request by davidyuk**
### Features

* export Encoding enum ([7ebc67f](https://github.com/aeternity/aepp-sdk-js/commit/7ebc67fdf32959960531cb96824fb3d181b99aed))

### Bug Fixes

* ability to use sdk without nodes ([8fa29e4](https://github.com/aeternity/aepp-sdk-js/commit/8fa29e4e81c6898573d259f15d8d9c0b1828ca94))
* **aepp,wallet:** avoid invalid message errors using third-party tools ([4d4d117](https://github.com/aeternity/aepp-sdk-js/commit/4d4d1175b539cb50bac277cb76ad5eefb9742a0d))
* **AeSdkBase:** don't mutate the options argument ([a574b7f](https://github.com/aeternity/aepp-sdk-js/commit/a574b7f00802b12334c3edc433fb0c53875c91cc))
* **awaitHeight:** allow to wait for arbitrary number of blocks ([918a885](https://github.com/aeternity/aepp-sdk-js/commit/918a8855f17c1a8fb8d4b6d1ffd930cd172ab429))
* **channel:** handle round as number ([97d7984](https://github.com/aeternity/aepp-sdk-js/commit/97d798438fb275974fa05bde5e52c21baf7e71b6))
* getNetworkId throw proper error without node ([3d9e73c](https://github.com/aeternity/aepp-sdk-js/commit/3d9e73c0fab6ecb40e027c68bf28859fc0127896))
* move type packages required by TS projects to dependencies ([90c5eb8](https://github.com/aeternity/aepp-sdk-js/commit/90c5eb8219d2a58c5760002ff947bcad8f578407))
* **node:** avoid ts-expect-error missed in generated d.ts file ([60d9755](https://github.com/aeternity/aepp-sdk-js/commit/60d97554a47e30483702e1b96c303e03417bc193))
* **node:** don't transform autorest properties ([7982327](https://github.com/aeternity/aepp-sdk-js/commit/7982327e1f33f915810ac94fd855baa8cfacb990))
* **Node:** tread `nameTtl`, `clientTtl` as number ([b82c398](https://github.com/aeternity/aepp-sdk-js/commit/b82c398cdda55b13749e31cd032818dc27c59449))
* **pollForQueryResponse:** poll until query is expired ([436f4a9](https://github.com/aeternity/aepp-sdk-js/commit/436f4a98b250e84fe47d7f2e0a5b578b690cf7c5))
* **sendTransaction:** don't fall if `onAccount` missed in AeSdkBase ([eebbcb7](https://github.com/aeternity/aepp-sdk-js/commit/eebbcb776b733d003bb138f891e0a7766edacb30))
* switch to aepp-calldata with fixed decoding of lists ([c1a3b24](https://github.com/aeternity/aepp-sdk-js/commit/c1a3b2458e30f2a16d6f1f33c0f5224471ba724d))

### CODE REVIEW

**Comment by davidyuk** [comment]
I'm planning to include #1627

File: src/chain.ts [hunk 1]
```diff
@@ -14,37 +14,148 @@
...
-  return Math.min(base / 3, _maxPollInterval)
+    microblock: _microBlockCycle,
+  }[type];
+  return Math.min(base / 3, _maxPollInterval);
+}
+
+/**
+ * @category exception
+ */
+export class InvalidTxError extends TransactionError {
```
Comment (Kalovelo) [inline src/chain.ts:53]: Wouldn't it be more concise to keep this under `utils/errors`?

File: src/deprecated/methods.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  { authData, authFun, ...options }: {
+    authData: Parameters<typeof createMetaTx>[1];
+    authFun: Parameters<typeof createMetaTx>[2];
+  } & Parameters<typeof createMetaTx>[3],
+): Promise<Encoded.Transaction> {
+  return createMetaTx(tx, authData, authFun, options);
+}
+
+/**
+ * @deprecated Use createMetaTx instead
```
Comment (Kalovelo) [inline src/deprecated/methods.ts:20]:
```suggestion
 * @deprecated Use getHeight instead
```

**Review by Kalovelo — COMMENTED** [review]
Congrats! Really like the new `Encoding`

File: src/chain.ts (hunk 1 above)
Comment (davidyuk) [reply src/chain.ts:53]: Sometimes errors (like InvalidTxError) depend on functionality or types defined in other modules, and keeping them in `utils/errors` leads to cycle dependencies that complicates testing and not working in some environments.

In general I'm trying to migrate from grouping functionality by programming entities (like `tx/builder/constants` and 
 `utils/errors`) to group it by problem it solves, for example https://github.com/aeternity/aepp-sdk-js/blob/4d203c6eecb525dfb8fa8ac84add2f607e77b1ee/src/tx/builder/field-types/fee.ts contains all functionality to calculate fee with minimal exports. Before it was splitted into constants (that actually not needed to other code) and a helper function that were all exported.

**Review by Kalovelo — APPROVED** [review]
Kudos! 🚀

**Pull request merged**